sql/core/src/main/scala/org/apache/spark/sql/internal/SessionState.scala

```diff
@@ -134,17 +131,20 @@ private[sql] class SessionState(
 
     SessionState.mergeSparkConf(confCopy, sparkContext.getConf)
 
-    new SessionState(
-      sparkContext,
-      newSparkSession.sharedState,
-      confCopy,
-      experimentalMethods.clone(),
-      functionRegistryCopy,
-      catalogCopy,
-      sqlParser,
-      SessionState.createAnalyzer(newSparkSession, catalogCopy, confCopy),
-      new StreamingQueryManager(newSparkSession),
-      queryExecutionCreator)
+    val newSessionState = new SessionState(
+        sparkContext,
+        newSparkSession.sharedState,
+        confCopy,
+        experimentalMethods.clone(),
+        functionRegistryCopy,
+        catalogCopy,
+        sqlParser,
+        SessionState.createAnalyzer(newSparkSession, catalogCopy, confCopy),
+        new StreamingQueryManager(newSparkSession),
+        queryExecutionCreator) {
+      override val listenerManager: ExecutionListenerManager = self.listenerManager.clone()
```

Comment:
I think the general rule we followed in the cloneSession PR is that `val`s that directly depend on `SparkSession` to be initialized, are to be constructor params. We leave the other `val`s in the body of class. 
The advantage here is less duplicated code between `SessionState`, `HiveSessionState` and `TestHiveSparkSession`.
This is consistent with that, what would be the advantage of changing it to be a param?